### PR TITLE
feat(processors.converter): convert tag or field as metric timestamp

### PR DESCRIPTION
**Pull request by powersj**
This adds the timestamp and timestamp_format options to the converter processor. This allows users to take a tag or field and make it the metric timestamp. It uses the same format parsing code that is used throughout Telegraf.

fixes: #12746

### CODE REVIEW

File: plugins/processors/converter/converter.go [hunk 1]
```diff
@@ -12,20 +12,23 @@ import (
...
-	Boolean     []string `toml:"boolean"`
-	Float       []string `toml:"float"`
+	Measurement     []string `toml:"measurement"`
+	Tag             []string `toml:"tag"`
+	String          []string `toml:"string"`
+	Integer         []string `toml:"integer"`
+	Unsigned        []string `toml:"unsigned"`
+	Boolean         []string `toml:"boolean"`
+	Float           []string `toml:"float"`
+	Timestamp       string   `toml:"timestamp"`
```
Comment (Hipska) [inline plugins/processors/converter/converter.go:30]: I would still use an array for uniformity and as its also done like this for the `measurement` name.

```suggestion
	Timestamp       []string `toml:"timestamp"`
```

File: plugins/processors/converter/sample.conf [hunk 2]
```diff
@@ -13,6 +13,13 @@
     boolean = []
     float = []
 
+    ## Optional tag to use as metric timestamp
+    # timestmap = ""
```
Comment (Hipska) [inline plugins/processors/converter/sample.conf:17]:
```suggestion
    float = []
    timestamp = []
```

File: plugins/processors/converter/sample.conf [hunk 3]
```diff
@@ -26,3 +33,11 @@
     unsigned = []
     boolean = []
     float = []
+
+    ## Optional field to use as metric timestamp
+    # timestmap = ""
```
Comment (Hipska) [inline plugins/processors/converter/sample.conf:38]:
```suggestion
    float = []
    timestamp = []
```

**Review by Hipska — CHANGES_REQUESTED** [review]

**Review by Hipska — APPROVED** [review]

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks for taking care of this @powersj!

**Pull request merged**